src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -44,13 +45,22 @@ AppHost::AppHost(const winrt::TerminalApp::AppLogic& logic,
 
     // _HandleCommandlineArgs will create a _windowLogic
     _useNonClientArea = _windowLogic.GetShowTabsInTitlebar();
-    if (_useNonClientArea)
+
+    const bool isWarmStart = window != nullptr;
+    if (isWarmStart)
     {
-        _window = std::make_unique<NonClientIslandWindow>(_windowLogic.GetRequestedTheme());
+        _window = std::move(window);

```

Comment:
OK so this was my fear. We always pop off whatever kind of window we're given, and we never make sure it's the right kind. That means that on Windows 11 if I change "Show Tabs in Titlebar" new windows will get it right (no refrigeration) and on Windows 10 it may get it wrong (it will microwave the wrong kind of window).

Reply:
On a scale of 1-blocking, where do we rate this?

Reply:
I would call it blocking for this PR.

Reply:
brainstorm
* keep two drawers in the fridge, one for each type of window. Evaluate what's needed, and get one of those specifically. 
  * seems insane
* Totally leak the fridge when that setting changes
  * will make sure we make the right kind of window
  * seems insane
  * leaks a bunch of memory when the setting changes
* Find some way to exorcise a DWXS out of a window when we change that setting
  * pretty sure the DWXS is HARD pinned to the `HWND`
  * yes it's HARD linked to the `HWND` [`IDesktopWindowXamlSourceNative::AttachToWindow`](https://learn.microsoft.com/en-us/windows/win32/api/windows.ui.xaml.hosting.desktopwindowxamlsource/nf-windows-ui-xaml-hosting-desktopwindowxamlsource-idesktopwindowxamlsourcenative-attachtowindow#remarks): 
    > Make sure that your code calls the AttachToWindow method only once per [DesktopWindowXamlSource](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.hosting.desktopwindowxamlsource) object. Calling this method more than once for a DesktopWindowXamlSource object could result in a memory leak
  * ... goddard, options

Reply:
Can't we just pin the setting for the entire duration the application is running? That's how the setting is documented anyways right?

Reply:
man I'd love that. That would sure be the easiest way.

Reply:
Er, after discussion, pinning probably isn't the best, cause warm-reloading `useTabsInTitlebar` _does_ work on Windows 11 today

Reply:
After discussing the impact with Mike, I'm choosing to not consider this blocking any longer.

It impacts the following scenario:

1. Be on Win 10
2. Open two windows
3. Close one
4. Change the "tabs in titlebar" setting
5. Open a new one